### PR TITLE
fix(storages): add check geo_restriciton in meta for private repo

### PR DESCRIPTION
**Pull request by ta4tsering**
check for geo_restricition in source_metadata, CN found it publish the repo as Private else it will be public repo.

### CODE REVIEW

**Comment by ta4tsering** [comment]
utils from openpecha is import from storage so it becoming circular import error https://github.com/OpenPecha/Toolkit/blob/fix-storages/openpecha/utils.py#L17

**Comment by kaldan007** [comment]
@10zinten publishing asset in release with pecha is also included in this request. did a dependency injection in epub serializer method to generalise css template

File: openpecha/core/pecha.py [hunk 1]
```diff
@@ -378,11 +382,22 @@ def reset_layers(self, base_name: str, exclude: List[LayerEnum] = []):
...
             self.storage = GithubStorage()
         self.storage.add_dir(path=self.pecha_path, description=self.about)
 
+        # Publishing assets in release
+        if asset_path:
+            repo_name = self.pecha_id
+            shutil.make_archive(asset_path.parent / asset_name, "zip", asset_path)
+            asset_paths.append(f"{asset_path.parent / asset_name}.zip")
+            create_release(
+                repo_name, prerelease=False, asset_paths=asset_paths, org=os.environ["OPENPECHA_DATA_GITHUB_ORG"], token=os.environ.get("GITHUB_TOKEN")
```
Comment (10zinten) [inline openpecha/core/pecha.py:397]:
```suggestion
                repo_name, prerelease=False, asset_paths=asset_paths, org=self.storage.org_name, token=self.storage.token)
```

File: openpecha/core/pecha.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import json
+import os
```
Comment (10zinten) [inline openpecha/core/pecha.py:2]:
```suggestion
```

required env var are read by `self.storage`

File: openpecha/core/pecha.py (hunk 1 above)
Comment (10zinten) [inline openpecha/core/pecha.py:397]:
```suggestion
                repo_name, prerelease=False, asset_paths=asset_paths, self.storage.org_name, token=self.storage.token)
```

use `self.storage` to get value for `org` and `token`

**Comment by kaldan007** [comment]
@10zinten suggestion considered and pecha about has been updated. Will send separate PR in future.

**Review by 10zinten — APPROVED** [review]

**Review by eroux — COMMENTED** [review]
just a quick note, we should check if `self.meta.source_metadata.get("geo_restriction", [])` is empty or not, any value should make the repository private, not just `"CN"`

**Comment by 10zinten** [comment]
> just a quick note, we should check if `self.meta.source_metadata.get("geo_restriction", [])` is empty or not, any value should make the repository private, not just `"CN"`

fixed with https://github.com/OpenPecha/Toolkit/pull/207

**Pull request merged**